libs-haskell/da-hs-base/src/DA/Signals.hs

```diff
@@ -25,3 +29,13 @@ installSignalHandlers = do
     mainThread <- myThreadId
     void $ installHandler sigTERM (Catch $ throwTo mainThread UserInterrupt) Nothing
 #endif
+
+withCloseOnStdin :: IO a -> IO a
+withCloseOnStdin a = do
+    mainThread <- myThreadId
+    withAsync (go mainThread) (const a)
```

Comment:
Lovely, like this it hangs on Windows since we cannot kill `isEOF` which I understand. If I change it to `(a >> hClose stdin)` as an attempt to fix this it even hangs on Linux for reasons that I do not understand. It is a mystery to me how anyone manages to write Haskell code that doesn’t end up locking itself up on Windows.

Reply:
Oh I understand what’s going on `hClose stdin` blocks because the handle is being used by `isEOF`.

Reply:
`const a >> getPid >>= terminateProcess` 😈

Reply:
I think I can write a loop around `hReady` to fix this slightly nicer :slightly_smiling_face:

Reply:
Tested the loop on Linux and Windows so this seems a workable solution. Tempted to start writing my own non-broken process library at this point.